### ISSUE TITLE
Download issue

### ISSUE THREAD

**Issue by 774254** [07th-mod/python-patcher#216]
When the installation program failed, there's a crash file hinting to submit it. The followings are error summary, please help me to solve this issue. 
Installer Version: [v1.2.16]
Thread panicked at 'Failed to send progress update - aborting extraction: SendError { .. }', src\archive_extractor.rs:127:14
-----------------------------------

   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: BaseThreadInitThunk
  14: RtlUserThreadStart

**Comment by 774254** [comment]
1 to 12 are followed by <unknown>

**Comment by 774254** [comment]
And there's an error: could not download [https://07th-mod.com/] with aria2c

**Comment by drojf** [comment]
Try the "nolauncher" version of the installer: https://github.com/07th-mod/python-patcher/releases/download/v1.2.20/07th-Mod.Installer.Windows.NoLauncher.zip

- Extract the `.zip` file to your desktop or downloads folder (NOT program files or the game root, as this may cause permissions issues)
- Double click on the `install.bat` file.
- A terminal will open, followed by a special page in your web browser (you use this webpage to install the mod).

![image](https://github.com/07th-mod/python-patcher/assets/1249449/789f4791-4e07-4bd1-abe7-4d3608bdd63f)

----

If it still does not work, please send the install log (drag and drop the file into Github):

![image](https://github.com/07th-mod/python-patcher/assets/1249449/165a1e42-a122-4d16-8322-c063d8586afb)


![image](https://github.com/07th-mod/python-patcher/assets/1249449/8bf474a8-f6d6-4a81-83e0-34c977f0127f)

![image](https://github.com/07th-mod/python-patcher/assets/1249449/a606de09-d46b-4e90-b49b-410a67e24fde)

**Comment by 774254** [comment]
[MOD-INSTALLER-LOG-2023-07-29_21-43-40.txt](https://github.com/07th-mod/python-patcher/files/12207297/MOD-INSTALLER-LOG-2023-07-29_21-43-40.txt)

**Comment by 774254** [comment]
It failed again with the same error.

**Comment by drojf** [comment]
hmm....I would say most likely your antivirus software is blocking the installer from accessing the internet, it looks the like installer can't access Github or 07th-mod. Are you running any additional antivirus software on your computer? You may need to unblock the installer inside the antivirus software.

We have a "General Connection Troubleshooting" page here: https://07th-mod.com/wiki/Installer/faq/#general-connection-troubleshooting which also states ad-blockers, anti-censorship software, and VPN software may interfere.

I'm assuming you have a working internet connection on the computer you're running the installer on - you need an internet connection to use the installer.

**Comment by drojf** [comment]
If you don't think any of your software is interfering, I can't think of much else to try - you probably just have to install manually: https://07th-mod.com/wiki/Higurashi/Manual-Installation/
https://07th-mod.com/wiki/Umineko/Umineko-Part-1---Voice-and-Graphics-Patch/

**Comment by 774254** [comment]
Ok, thank you very much!!!

**Comment by drojf** [comment]
Actually, I forgot I had this prototype version of the installer which you might be able to use. It works even if the installer can't access the internet, but you have to download the files manually yourself.

You can download it from here: [07th-Mod.Installer.Windows.NoLauncher.v1.2.19-offline-test.zip](https://github.com/07th-mod/python-patcher/files/12208426/07th-Mod.Installer.Windows.NoLauncher.v1.2.19-offline-test.zip)

Please ignore the "outdated installer version" warning.

**You MUST enable the option "Download Manually" in the installer to use this prototype installer, see pic below**

There are some instructions on using this "Download Manually" option here; https://07th-mod.com/wiki/Installer/faq/#workaround-for-download-and-verify-stage-failed-error and also inside the installer itself

![image](https://github.com/07th-mod/python-patcher/assets/1249449/ded5d53b-794a-4ef0-8762-20bea9a72945)

**Comment by 774254** [comment]
![image](https://github.com/07th-mod/python-patcher/assets/140897980/9917b0c8-7b35-4dc8-98df-7119b8d85daa)
It worked! But it stuck here, always validating.

**Comment by drojf** [comment]
Can you send the installer log again?

**Comment by 774254** [comment]
[MOD-INSTALLER-LOG-2023-07-30_11-39-39.txt](https://github.com/07th-mod/python-patcher/files/12208470/MOD-INSTALLER-LOG-2023-07-30_11-39-39.txt)

**Comment by drojf** [comment]
Can you try this new version of the installer?

[07th-Mod.Installer.Windows.NoLauncher.v1.2.19c-offline-test.zip](https://github.com/07th-mod/python-patcher/files/12208516/07th-Mod.Installer.Windows.NoLauncher.v1.2.19c-offline-test.zip)

**Comment by 774254** [comment]
![image](https://github.com/07th-mod/python-patcher/assets/140897980/7bd8e28a-dcfd-42d2-a6f0-ed59f5840e0d)
[MOD-INSTALLER-LOG-2023-07-30_12-36-35.txt](https://github.com/07th-mod/python-patcher/files/12208538/MOD-INSTALLER-LOG-2023-07-30_12-36-35.txt)

Yes, I can see the page of download manually now. But when I clicked that button, the installation failed again. Maybe I could find a cafe that I can use its wifi and all problems can be solved. Anyway, thanks again.

**Comment by 774254** [comment]
Now I have installed the required plug manually in the 07th-mod web. There's just BGM and no characters' voices in the game. So I compared the settings of the game (by the way, it's the third game) with the second game with characters' voices installed by automatic installation, and settings are the same, but why it's still no voices....

**Comment by drojf** [comment]
Does pressing F10 open the mod menu?

Does everything else work? Can you see the console backgrounds and sprites?

Does your config screen look like this?
![image](https://github.com/07th-mod/python-patcher/assets/1249449/5cb6c16b-dfb6-43e8-ad20-35feaff02014)


Do you  have a voice folder like this?

![image](https://github.com/07th-mod/python-patcher/assets/1249449/73572d9d-892f-4233-bef3-e5ad78a757b8)

![image](https://github.com/07th-mod/python-patcher/assets/1249449/c35c0aad-3d89-44bb-be82-b698bd0847aa)

![image](https://github.com/07th-mod/python-patcher/assets/1249449/f0e3206b-72c7-47d9-b169-479860cfbcb0)
![image](https://github.com/07th-mod/python-patcher/assets/1249449/618d2826-5a72-477d-8162-2ac158813db8)
![image](https://github.com/07th-mod/python-patcher/assets/1249449/4ef81f6d-1053-48fe-a233-f87486f0eb3b)

**Comment by drojf** [comment]
Also, I'm pretty sure the installer worked OK, you just didn't put the files in the specified folder:

![image](https://github.com/07th-mod/python-patcher/assets/1249449/694235b0-73d9-4c79-8442-83b878f761e0)

You need to manually download the files (you've already done this), then put them inside the specified folder.

Then, when all the rows turn green, you can start the installer.

**Comment by 774254** [comment]
3yes, but the voice folder is obviously incomplete, it must be the reason. I'll go search for the complete edition. 
![image](https://github.com/07th-mod/python-patcher/assets/140897980/690e1720-74a6-4d51-b555-6cd7e3e8527d)

**Comment by 774254** [comment]
Finally I can hear the voice, but suddenly the graphics disappeared, like this
![image](https://github.com/07th-mod/python-patcher/assets/140897980/04989602-0ceb-4855-835b-58072ef8172d)

**Comment by 774254** [comment]
It's in campus, so the background  oughts to be not black

**Comment by 774254** [comment]
wait, it's ok 
![image](https://github.com/07th-mod/python-patcher/assets/140897980/edaeaba0-05a4-494f-9506-88a08a9fd9c5)

**Comment by 774254** [comment]
I have no more issues for now, thank for your time and have a wonderful day (though I don't know it's day or night there)
